orttraining/orttraining/test/python/checkpoint/_test_helpers.py

```diff
@@ -102,6 +104,74 @@ def create_orttrainer_and_load_checkpoint(device, trainer_opts, checkpoint_dir,
+
+def verify_model_state(trainer, expected_state_dict, is_mixedprecision):
+    actual_model_state = trainer._training_session.get_model_state(include_mixed_precision_weights=True)
+    
+    assert len(expected_state_dict['fp32_param']) == len(actual_model_state['full_precision']), \
+        "expected and actual should have same number of tensors"
+    for weight_name, tensor in expected_state_dict['fp32_param'].items():
+        if not weight_name in actual_model_state['full_precision']:
+            assert '_view_' in weight_name, \
+                "only zero shared weight may not match name"
+            weight_name = weight_name.split('_view_')[0]
+        assert_allclose(tensor, actual_model_state['full_precision'][weight_name])
+
+    if is_mixedprecision:
+        assert 'mixed_precision' in actual_model_state.keys(), "missing 'mixed_precision' key in mixed precision run"
+        assert len(expected_state_dict['fp16_param']) == len(actual_model_state['mixed_precision']), \
+            "expected and actual should have same number of tensors"
+        for weight_name, tensor in expected_state_dict['fp16_param'].items():
+            weight_name = weight_name.split('_fp16')[0]
+            assert_allclose(tensor, actual_model_state['mixed_precision'][weight_name])
+
+def verify_opt_state(trainer, expected_state_dict):
+    actual_opt_state = trainer._training_session.get_optimizer_state()
+    actual_opt_count = sum(len(v) for v in actual_opt_state.values())
+    assert actual_opt_count == len(expected_state_dict['optimizer'])
+    for opt_name, expected_tensor in expected_state_dict['optimizer'].items():
+        if opt_name == "Step":
+            actual_tensor = actual_opt_state['shared_optimizer_state']['Step']
+        else:
+            if opt_name.startswith('Moment_'):
+                prefix = opt_name[:len("Moment_0")]
+                weight_name = opt_name[len("Moment_0_"):]
+                if not weight_name in actual_opt_state:
+                    assert '_view_' in weight_name, \
+                        "only zero shared weight may not match name"
+                    weight_name = weight_name.split('_view_')[0]
+            elif opt_name.startswith('Update_Count_'):
+                prefix = "Update_Count"
+                weight_name = opt_name[len(prefix + 1):]
+            actual_tensor = actual_opt_state[weight_name][prefix]
+        assert_allclose(actual_tensor, expected_tensor, atol=global_fp16_fp32_atol)
+
+def verify_part_info(trainer, expected_state_dict, is_zero_run):
+    part_info = trainer._training_session.get_partition_info_map()
+    for weight_name, weight_info in part_info.items():
+        for info, value in weight_info.items():
+            assert isinstance(value, list), "get_partition_info_map should return list"
+            assert isinstance(value[0], int), "get_partition_info_map should return list of int"
+            if is_zero_run:
+                if info == "megatron_row_partition":
```

Comment:
assert that it is -1 for megatron unused

Reply:
assert megatron_row_partition is -1 now